nep-22.mediawiki

```diff
@@ -0,0 +1,67 @@
+<pre>
+  NEP: 22
+  Title: Contract Update Function
+  Author: Owen Zhang <zhangzhihong@ngd.neo.org>, Fernando Díaz Toledano <shargon@gmail.com>, Erik Zhang <erik@neo.org>, Roman Khimov <roman@nspcc.ru>
+  Type: Standard
+  Status: Accepted
+  Created: 2021-04-19
+</pre>
+
+
+==Abstract==
+
+This proposal outlines a standard for the NEO blockchain that will provide systems with a generalized interaction mechanism for smart contract update.
+
+==Motivation==
+
+As the NEO blockchain scales, Smart Contract Update will become increasingly important. We need standize <code>update</code> method in contract to callback native <code>ContractManagement</code> correctly.
+
+==Specification==
+
+Neo N3 has a native <code>ContractManagement</code> contract that is used for contract deployment and update via <code>deploy</code> and <code>update</code> methods of it. After settling with the new NEF and manifest both of them will invoke a special <code>_deploy</code> method if it's implemented by the contract. More details in [NEP-29](https://github.com/neo-project/proposals/blob/master/nep-29.mediawiki)
+
+===Methods===
+
+====update====
+
+<pre>
+{
+  "name": "update",
+  "safe": false,
+  "parameters": [
+    {
+      "name": "nefFile",
+      "type": "ByteArray"
+    },
+    {
+      "name": "manifest",
+      "type": "ByteArray"
+    },
+    {
+      "name": "data",
+      "type": "Any"
+    }
+  ],
+  "returntype": "Void"
+}
+</pre>
+
+This method will will callback <code>Update</code> method in native <code>ContractManagement</code> contract. It will update contract state if it's exectued successfully.
+
+Updating a smart contract MUST have <code>nefFile</code> or <code>manifest.json</code>. If just update one of them at a time is OK, another SHOULD be set as null. It will be passed to <code>_deploy</code>.
```

Comment:
Maybe refer to NEP-16 and NEP-15 here. I also don't understand 

> It will be passed to <code>_deploy</code>.

here. Only `data` is going to be passed and that's exactly where a reference to NEP-29 is needed.